### ISSUE TITLE
Explore making BloodHound accessible to screen reader users

### ISSUE THREAD

**Issue by zersiax** [SpecterOps/BloodHound#575]
**Feature Description:**

Given the web/browser-based nature of the UI, we have a rich set of APIs available to us in order to render content in a way that is semantic and according to established accessibility standards (WCAG 2.2 etc.)
While a 100% conformance to WCAG 2.2 AA is lofty, in particular a textual/tabular alternative to currently graph-only data would already help immensely in making this tool usable for blind/visually impaired cybersecurity professionals, students, etc.

**Current Behavior:**

Rather a lot of data Bloodhound provides on given data is, at least for as far as I can determine, only available as a graph. While this is great for getting at-a-glance information on the state of things, people who can't see or interpret the graph are essentially blocked completely on using the tool effectively, if at all. WCAG generally tends to recommend always have more than one way to indicate important info (e.g. an image + alt text for an image, color + error message for an incorrectly filled in form field etc.), and the spirit of this feature request is to work towards doing this for currently " visual-only"  information as well.

**Desired Behavior:**

Using some kind of secondary representation of this data, enable non-visual users to still use this tool. Some examples:

- Table/definition list/heading-separated text render of datapoints indicated on the graph
- Accessible graphing library (unfortunately the only one I know if HighCharts which I am pretty sure is non-free),
- For parent-child relationships, a treeview pattern might be of use, e.g. https://www.w3.org/WAI/ARIA/apg/patterns/treeview/examples/treeview-navigation/ . My exposure to the software is limited, and therefore I am not sure how applicable this would be.
- Failing all that, some kind of JSON export of the graph's datapoints so a user can import into homebrew tools in order to further analyze (e.g. jq). Maybe this already exists?

 

**Use Case:**

Right now we're excluding a bunch of users who might otherwise be helped by this tool. By doing this, we stop doing that.

**Implementation Suggestions:**

See above.
